blockmanager.go

```diff
@@ -3044,3 +3091,167 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
 		ancestorHeight, ancestor, l.store, l.headerList,
 	)
 }
+
+func (b *blockManager) sideLoadHeaders() {
+	// If sideloading is not enabled on this chain
+	// return quickly.
+	if !b.cfg.sideLoad.Enabled {
+		return
+	}
+
+	// If headers contained in the side load source are for a different chain network return
+	// immediately.
+	if b.sideLoadBlkHdrReader.HeadersChain() != b.cfg.ChainParams.Net {
+		log.Error("headers from side load file are of network %v "+
+			"and so incompatible with neutrino's current bitcoin network "+
+			"-- skipping side loading", b.sideLoadBlkHdrReader.HeadersChain())
+
+		return
+	}
+	log.Infof("Side loading headers from %v to %v", b.headerTip, b.sideLoadBlkHdrReader.EndHeight())
+
+	// Set headerTip to enable reader supply header, node needs
+	err := b.sideLoadBlkHdrReader.SetHeight(int64(b.headerTip))
+	if err != nil {
+		log.Errorf("error while setting height for sideload--- skipping sideloading: "+
+			"%v", err)
+
+		return
+	}
+
+	// Start batch writer to write headers into DB at specified batch size and ticker duration.
+	b.blockHeaderBatchWriter.Start()
+	defer b.blockHeaderBatchWriter.Stop()
```

Comment:
See above, we can just write this in one swoop.

Reply:
Depend on this: https://github.com/lightninglabs/neutrino/pull/285#discussion_r1375169669